python/cudf/cudf/core/frame.py

```diff
@@ -1489,6 +1486,13 @@ def casting_rules(lhs, rhs, dtype_l, dtype_r, how):
             return rtn
 
         if left_index or right_index:
+            if isinstance(
+                lhs.index, cudf.core.multiindex.MultiIndex
+            ) or isinstance(rhs.index, cudf.core.multiindex.MultiIndex):
+                warnings.warn(
+                    "Implicit typecasting of multi-indices not yet supported"
+                )
```

Comment:
What happens without this warning? Does something throw down the line due to mismatched dtypes? It would be better to just throw where appropriate with a `NotImplementedError` for now than to always warn when joining on MultiIndexes.

Reply:
Libcudf will error with a dtype mismatch error I believe. This should be changed though, I agree.